File: plugin/email/www/email.js
window.plugin = {
  name: 'Email',
  loaded: 'email_plugin_loaded'
};


function email_plugin_loaded() {

	$('#EmailCancelButton').click(function(){
		getEmailData();
		return false;
	});

	$('#EmailSaveButton').click(function(){

		$("body").css("cursor", "progress");

		var dataString = 'active=' + $("#Active").prop('checked') +
		'&recipient=' + encodeURIComponent($('#Recipient').val()) +
		'&subject=' + encodeURIComponent($('#Subject').val()) +
		'&message=' + encodeURIComponent($('#Message').val()) +
		'&attach=' + $("#Attach").prop('checked') +
		'&smtp=' + encodeURIComponent($('#Smtp').val()) +
		'&user=' + encodeURIComponent($('#User').val()) +
		'&password=' + encodeURIComponent($('#Password').val());
		$.ajax({
			type: "POST",
			url: "/plugin/email/email.py/set_email_data",
			data: dataString,
			dataType: "json",
			success: function(response) {
				$("body").css("cursor", "default");
				if (response.Result == 'ERROR')
					alert(response.Message);
			},
			error: function (xhr, ajaxOptions, thrownError) {
				$("body").css("cursor", "default");
				alert('Error: ' + xhr.status + ' - ' + thrownError);
			},
		});
		return false;
	});

	getEmailData();
}


function getEmailData() {
    $.ajax({
        type: "GET",
        url: "/plugin/email/email.py/get_email_data",
        contentType: "application/json; charset=utf-8",
        success: function(response){
			if (response.Result == 'ERROR')
				alert(response.Message);
			else {
	        	$("#Active").prop('checked', (response.Active == 'true'));
	        	$("#Recipient").val(response.Recipient);
	        	$("#Subject").val(response.Subject);
	        	$("#Message").val(response.Message);
	        	$("#Attach").prop('checked', (response.Attach == 'true'));
	        	$("#Smtp").val(response.Smtp);
	        	$("#User").val(response.User);
	        	$("#Password").val(response.Password);
    		}
        },
    });
}

File: plugin/email/www/email.py
# Watchdog - Building supervision with Raspberry Pi
# (C)2017  - Norbert Huffschmid - GNU GPL V3 

import ConfigParser
import json
import re

CONFIG_FILE = '/etc/watchdog.conf'


def get_email_data(req):
    req.content_type = 'application/json; charset=UTF8'

    try:
        result = { 'Result': 'OK'}

        config = ConfigParser.SafeConfigParser()
        config.read(CONFIG_FILE)
        result['Active'] = config.get('email', 'active')
        result['Recipient'] = config.get('email', 'recipient')
        result['Subject'] = config.get('email', 'subject')
        result['Message'] = config.get('email', 'message')
        result['Attach'] = config.get('email', 'attach')
        result['Smtp'] = config.get('email', 'smtp')
        result['User'] = config.get('email', 'user')
        result['Password'] = config.get('email', 'password')

        return json.dumps(result)

    except Exception, e:
        return json.dumps({ 'Result': 'ERROR', 'Message': 'Cannot get email data: ' + str(e) })


def set_email_data(req):
    req.content_type = 'application/json; charset=UTF8'

    try:

        # read POST parameters 
        active = req.form.getfirst('active')
        recipient = req.form.getfirst('recipient')
        subject = req.form.getfirst('subject')
        message = req.form.getfirst('message')
        attach = req.form.getfirst('attach')
        smtp = req.form.getfirst('smtp')
        user = req.form.getfirst('user')
        password = req.form.getfirst('password')
        
        # validity check for recipient(s)
        valid_email = '[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]+'
        if not re.match('^' + valid_email + '(;' + valid_email + ')*$', recipient):
            raise ValueError('Invalid recipient(s):\n' + recipient)

        config = ConfigParser.SafeConfigParser()
        config.read(CONFIG_FILE)
        config.set('email', 'active', active)
        config.set('email', 'recipient', recipient)
        config.set('email', 'subject', subject)
        config.set('email', 'message', message)
        config.set('email', 'attach', attach)
        config.set('email', 'smtp', smtp)
        config.set('email', 'user', user)
        config.set('email', 'password', password)
        with open(CONFIG_FILE, 'wb') as configfile:
            config.write(configfile)

        return json.dumps({ 'Result': 'OK' })

    except Exception, e:
        return json.dumps({ 'Result': 'ERROR', 'Message': 'Cannot set email data: ' + str(e) })
